src/spec-configuration/containerCollectionsOCI.ts

```diff
@@ -309,11 +310,17 @@ export async function getManifest(params: CommonParams, url: string, ref: OCIRef
 	// That is useful to have, so if the server doesn't provide it, recalculate it outselves.
 	// Headers are always automatically downcased by node.
 	let contentDigest = headers['docker-content-digest'];
-	if (!contentDigest) {
-		output.write('Registry did not send a \'docker-content-digest\' header.  Recalculating...', LogLevel.Trace);
+	if (!contentDigest || expectedDigest) {
+		if (!contentDigest) {
+			output.write('Registry did not send a \'docker-content-digest\' header.  Recalculating...', LogLevel.Trace);
+		}
 		contentDigest = `sha256:${crypto.createHash('sha256').update(body).digest('hex')}`;
 	}
 
+	if (expectedDigest && contentDigest !== expectedDigest) {
+		throw new Error(`Digest did not match for ${url}.`);
```

Comment:
Maybe use the `ref.resource` here instead of URL for a more clear error message?